engines/mxnet/mxnet-engine/build.gradle

```diff
@@ -32,7 +30,14 @@ pmdMain.source = 'src/main/java'
 
 tasks.register('jnarator') {
     dependsOn ":engines:mxnet:jnarator:jar"
+    //declare inputs and outputs to allow Gradle to cache the task or skip it if inputs do not change
+    inputs.file("${project.projectDir}/src/main/jna/mapping.properties")
+        .withPathSensitivity(PathSensitivity.RELATIVE)
+        .withPropertyName("jna/mapping.properties file")
     outputs.dir "${project.buildDir}/generated-src"
+    //there is no reason not to cache the task if Gradle needs to, so enable it
```

Comment:
nit: I think you can remove this comment

Reply:
i agree we can remove this. how long does this task take to execute?